blogposts/inside-rust/01-announcement.md

```diff
@@ -0,0 +1,166 @@
+defined by Rust, should permit unwinding.
+
+This is not a question we expected to be debating. We've long declared that
+unwinding through Rust's `"C"` ABI is undefined behavior. In part, this is
+because nobody had spent the time to figure out what the correct behavior would
+be, or how to implement it, although (as we'll see shortly) there are other
+good reasons for this choice. 
+
+In any case, in PR #65646, @Amanieu proposed that we could, in fact, simply
+define the behavior of unwinding across `"C"` boundaries. In discussing this,
+discovered that the question of whether the `"C"` ABI should permit unwinding was
+less clear-cut than we had assumed.
+
+If the `"C"` ABI does not permit unwinding, a new ABI, called `"C unwind"`,
+will be introduced specifically to support unwinding.
+
+## Three specific proposals
+
+The project group has narrowed the design space down to three specific
+proposals. Two of these introduce the new `"C unwind"` ABI, and one does not.
+
+Each proposal specifies the behavior of each type of unwind (Rust `panic!`,
+foreign (e.g. C++), and forced (e.g. `pthread_exit`)) when it encounters an
+ABI boundary under either the `panic=unwind` or `panic=abort` compile-mode.
+
+### 2 ABIs, minimal specification
+
+* `"C"` ABI boundary, `panic=<any>`
+  * `panic!`: program aborts
+  * forced unwind, no destructors: unwind behaves normally
+  * all other foreign unwinding: undefined behavior
+* `"C unwind"` ABI boundary
+  * With `panic=unwind`: all types of unwinding behave normally
+  * With `panic=abort`: all types of unwinding abort the program
+
+### 2 ABIs, forced unwinding always permitted
+
+This is the same as the previous design, except that when compiled with
+`panic=abort`, forced unwinding would *not* be intercepted at `"C unwind"` ABI
+boundaries; that is, they would behave normally, without causing the program to
+abort. `panic!` and non-forced foreign exceptions would still cause the program
+to abort.
+
+### 1 ABI
+
+* `panic=unwind`: unwind behaves normally
+* `panic=abort`:
+  * `panic!`: program aborts
+  * forced unwind, no destructors: unwind behaves normally
+  * all other foreign unwinding: undefined behavior
+
+## Comparison table for the proposed designs
+
+In this table, "UB" stands for "undefined behavior". Some instances of UB can
+be detected at runtime, but the code to do so would impose an undesirable
+code-size penalty; the group recommends generating such code only for debug
+builds. These cases are marked "UB (debug: abort)"
+
+Note that forced unwinding always causes undefined behavior when it traverses
+stack frames with destructors.
+
+|                                                         | “2 ABIs”, “C”, “panic=unwind” | “2 ABIs”, “C”, “panic=abort” | “2 ABIs”, “C unwind”, “panic=unwind” | “2 ABIs, always permit forced”, “C unwind”, “panic=abort” | “2 ABIs, minimal spec”, “C unwind”, “panic=abort” | “1 ABI”, “panic=unwind” | “1 ABI”,  “panic=abort” |
```

Comment:
I think this table would be more readable if we rotated it, so that the *columns* are the kind of unwinding (panic, forced unwind, etc) and the *rows* are the specific scenarios and proposals. It's quite wide as is.

Reply:
Done! Good suggestion.